common/src/main/java/com/wynntils/models/tracker/TrackerScoreboardPart.java

```diff
@@ -45,24 +46,31 @@ public void onSegmentChange(ScoreboardSegment newValue) {
             }
         }
 
+        String unformattedHeader = newValue.getHeader().getString(PartStyle.StyleType.NONE);
+        Matcher matcher = TRACKER_MATCHER.headerPattern().matcher(unformattedHeader);
+
+        assert matcher.matches();
```

Comment:
Qodana is right? I am not sure how asserts work in PROD, but if they are completely ignored, then this matches call wouldn't happen, would it?

Reply:
I can't see Qodana's comment..? The assert will be ignored unless you run with asserts enabled. We do that in our IntelliJ profile but not in production. We know that the matcher is going to match since we only get a callback here if it matched in the caller. So this assert was more to show that I know this. Otherwise it might look suspicious to just check the group.

Reply:
Now I saw it. I tried to read the docs for matches() and group(), and it seems that they say that I must call matches() before group(), so I guess this would have broken in prod.